server/container_create.go

```diff
@@ -286,9 +313,18 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 
 	logrus.Debugf("pod container state %+v", podInfraState)
 
-	ipcNsPath := fmt.Sprintf("/proc/%d/ns/ipc", podInfraState.Pid)
-	if err := specgen.AddOrReplaceLinuxNamespace("ipc", ipcNsPath); err != nil {
-		return nil, err
+	for nsType, nsFile := range map[rspec.NamespaceType]string{
+		rspec.PIDNamespace:    "pid",
+		rspec.MountNamespace:  "mount",
+		rspec.IPCNamespace:    "ipc",
+		rspec.UTSNamespace:    "uts",
+		rspec.UserNamespace:   "user",
+		rspec.CgroupNamespace: "cgroup",
+	} {
+		nsPath := fmt.Sprintf("/proc/%d/ns/%s", podInfraState.Pid, nsFile)
+		if err := specgen.AddOrReplaceLinuxNamespace((string)(nsType), nsPath); err != nil {
+			return nil, err
+		}
```

Comment:
That nsPath looks like it should use filepath.Join instead? Also `string(nsType)` instead of `(string)(nsType)`? (I didn't even realize the latter is even valid Go code?)